src/components/Login/Login.test.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+
+import { loginModalClosed, loginModalOpened } from '../../__mocks__/loginMock';
+import { useAuth } from '../../providers/Auth';
+import Login from './Login.component';
+
+jest.mock('../../providers/Auth', () => ({
+  useAuth: jest.fn(() => ({ authenticated: false })),
+}));
+
+describe('Login', () => {
+  it('should renders when modal is opened', () => {
+    const { getByTestId } = render(<Login {...loginModalOpened} />);
+
+    expect(getByTestId('login-modal')).toBeInTheDocument();
+  });
+
+  it('should not renders when modal is closed', () => {
+    const { queryByTestId } = render(<Login {...loginModalClosed} />);
+
+    expect(queryByTestId('login-modal')).not.toBeInTheDocument();
+  });
+
+  it('should allows user to login', () => {
+    useAuth.mockImplementation(() => ({ authenticated: true, login: jest.fn() }));
+    const { getByTestId } = render(<Login {...loginModalOpened} />);
+
+    const usernameInput = getByTestId('login-username-input');
+    const passwordInput = getByTestId('login-password-input');
+
+    fireEvent.change(usernameInput, { target: { value: 'admin' } });
+    fireEvent.change(passwordInput, { target: { value: 'admin' } });
+    expect(usernameInput.value).toBe('admin');
+    expect(passwordInput.value).toBe('admin');
+
+    fireEvent.click(getByTestId('login-button'));
+    expect(usernameInput.value).toBe('');
+    expect(passwordInput.value).toBe('');
+  });
```

Comment:
Suggestion:
- another good expectation would be to check that the login fn was called once

Reply:
Sure, I will add it. Thanks!